### ISSUE TITLE
bug(Datepicker): Accessibility screen reader identifies incorrect number of rows in calendar.

### ISSUE THREAD

**Issue by amaust82** [angular/components#28476]
### Is this a regression?

- [ ] Yes, this behavior used to work in the previous version

### Description

When using a keyboard while running a screen reader such as NVDA, upon opening the calendar the screen reader reports seven rows, even if there is only 6 visually. 

### Reproduction

Steps to reproduce:

1. launch a screen reader such as JAWS or NVDA
2. go to [https://material.angular.io/components/datepicker/overview](https://material.angular.io/components/datepicker/overview)
3. tab to the "Basic datepicker" field's calendar icon and hit ENTER to launch it
4. Navigate to Feb 2024
5. The screen reader narrates "table with 7 rows and 7 columns row 1 column 1 Sunday" yet visually 6 rows are provided in the calendar table. 

### Expected Behavior

The screen reader should narrate the correct number of rows, 6 (Header + calendar days)

### Actual Behavior

The screen reader narrates 7 rows, even though there are 6

### Environment

- Angular: 16
- CDK/Material: 16
- Browser(s): Chrome
- Operating System (e.g. Windows, macOS, Ubuntu): Windows 11

**Comment by amaust82** [comment]
Forgot to mention, in the code it looks like the divider row has aria-hidden on the table cell with a colspan of 7, but I'm wondering if that should be moved up to the <tr> tag above it.

**Comment by zarend** [comment]
@essjay05 : this would be a good one for you to look at when you are available to. The fix may be as simple as putting `aria-hidden="true"` on the divider row 🤞
